### ISSUE TITLE
Native Go ping requires `CAP_NET_RAW` but Telegraf binary doesn't have capability set in Docker image

### ISSUE THREAD

**Issue by Cougar** [influxdata/influxdata-docker#547]
This is basically a copy of influxdata/telegraf issue [#10069](https://github.com/influxdata/telegraf/issues/10069). I hope that this is the better place to report and discuss Docker behavior.

Problem with Docker image is that after [#11200](https://github.com/docker-library/official-images/pull/11200) and [#11226](https://github.com/docker-library/official-images/pull/11226) it runs under non-privileged user (very good!) but now it needs at least `CAP_NET_RAW` capability to execute native builtin ping plugin.

I read [blog post](https://www.influxdata.com/blog/docker-run-telegraf-as-non-root/) about moving away from root privilege but there was no word about capabilities and policy about that.

Are these "bad" as well or can we consider using telegraf under normal user but adding some capabilities to give it more permissions? My proposal is for a change like this:

```diff
diff --git a/telegraf/1.20/alpine/Dockerfile b/telegraf/1.20/alpine/Dockerfile
index 42c084e..95e6b9b 100644
--- a/telegraf/1.20/alpine/Dockerfile
+++ b/telegraf/1.20/alpine/Dockerfile
@@ -23,6 +23,7 @@ RUN set -ex && \
     mv /usr/src/telegraf*/etc/telegraf/telegraf.conf /etc/telegraf/ && \
     mkdir /etc/telegraf/telegraf.d && \
     cp -a /usr/src/telegraf*/usr/bin/telegraf /usr/bin/ && \
+    setcap cap_net_raw+ep /usr/bin/telegraf && \
     gpgconf --kill all && \
     rm -rf *.tar.gz* /usr/src /root/.gnupg && \
     apk del .build-deps && \
```

Or is the intention that in case such permissions are needed, the end user should always build its own image based on official image and add necessary capabilities like this?

```Dockerfile
FROM telegraf:alpine
RUN setcap cap_net_raw+ep /usr/bin/telegraf
```

There is no way to give this permission just by running current image (using `--cap-add NET_RAW` for example).

**Comment by Hipska** [comment]
Could we in the meantime also request to have `net_bind_service` as well?

**Comment by Hipska** [comment]
@powersj @reimda Tagging you guys here..

**Comment by loganmc10** [comment]
When the 1.20.3 image came out, I switched to using "runas" to continue to run as root as a temporary workaround (in Kubernetes).

I just tried the 1.20.4 image and I get:
```
Error running agent: starting input inputs.snmp_trap: listen udp :162: bind: permission denied
```
Even when running as root.

Also, I'd like to mention: breaking changes like this should make it into the main Telegraf release notes. Most people don't expect this kind of thing when updating a point release, like 1.20.3 to 1.20.4, especially when it's not mentioned anywhere in the release notes
